bidfx/pricing/_pixie/util/varint.py

```diff
@@ -96,7 +96,12 @@ def decode_varint_from_socket(socket_connection):
 
 
 def read_bytes(socket_connection, length):
-    byte_ = socket_connection.recv(length)
-    if byte_ == b"":
-        raise socket.error("end of socket stream")
-    return byte_
+    result = b""
+
+    while len(result) != length:
```

Comment:
Although not wrong, I don't like the use of != as there could be a potential error left uncaught should you run past the length. Better to explicitly use < as an extra guard against an infinite loop.